### PR TITLE
[NCL-2666] Switch DA CLI to websocket endpoints for reports

### PR DESCRIPTION
**Pull request by stanlyDoge**
skip ci

### CODE REVIEW

**Comment by janinko** [comment]
I think you would like to rebase so there are not the "add client autehntication" and "Rewrite CLI tool ..." commits.

**Comment by janinko** [comment]
You mix python2 and python3? Why does the files have different shebangs?

File: cli-wrap/src/main/resources/cli/listings.py [hunk 1]
```diff
@@ -406,13 +456,26 @@ def scm_report_adv():
...
+            await websocket.send(query)
+            output = await websocket.recv()
+        try:
+            output = json.loads(output)["result"]
+        except KeyError:
+            print(json.loads(output)["error"]["message"])
+            exit()
+    except websockets.exceptions.InvalidHandshake:
+        print("Server " + "ws://"+da_cli_script.da_server_ws + " not found!")
+        exit()
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.py]: This seems repeted a lot, can't it be extracted?

File: cli-wrap/src/main/resources/cli/listings.py [hunk 2]
```diff
@@ -157,7 +159,7 @@ def pom_bw_junit_xml(argv, parser):
     
 def formatGAVjsonRequest(groupId, artifactId, version):
     return "\"groupId\":\""+ groupId + "\", \"artifactId\":\""+ artifactId +"\", \"version\":\""+ version  +"\""
-    
+
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.py]: whitespace

File: cli-wrap/src/main/resources/cli/listings.py [hunk 3]
```diff
@@ -512,10 +512,25 @@ def prod_difference():
...
-            print output['errorMessage']
+            print(output['errorMessage'])
             exit()
         pretty_out = pretty.difference(output)
         print(pretty_out)
         exit()
     else:
         print(json.dumps(output))
+              
+async def get_response(query):
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.py]: Why is the method async?

Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/listings.py]: output is future object and you have to wait until it is complete -> output = asyncio.get_event_loop().run_until_complete(get_response(query))

File: cli-wrap/src/main/resources/cli/listings.py (hunk 1 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/listings.py]: done

File: cli-wrap/src/main/resources/cli/da_cli_script.py [hunk 4]
```diff
@@ -10,12 +10,16 @@
...
 
-if (da_server == ""):
-    print("Please configure DA server by command $ export DA_SERVER=your.adress/rest/v-1")
+if (da_server_raw == ""):
+    print("Please configure DA server by command $ export DA_SERVER=your.adress")
     print("or by filling address in configuration file (config.json).")
     exit()
+    
+da_server = da_server_raw + "/da/rest/v-1"
+da_server_ws = da_server_raw + "/da/ws"
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da_cli_script.py]: I realized that REST uses HTTP protocol and websockets uses WS protocol. The url in config should be protocol less and you should add the protocol here.

File: cli-wrap/src/main/resources/cli/da-cli.py [hunk 5]
```diff
@@ -339,6 +339,15 @@ def list_black(self):
...
             
+    def list_white(self):
+        if len(sys.argv) == 3:
+            self.da.print_white_artifacts()
+        elif len(sys.argv) == 4:
+             if self.pm.matchProd(sys.argv[3]):
+                self.da.print_white_artifacts(sys.argv[3])
+        else:
+            print("Bad arguments! For help use: ./da-cli.py --help")
+
```
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.py]: Why this appeared here?

File: cli-wrap/src/main/resources/cli/listings.py (hunk 3 above)
Comment (janinko) [inline cli-wrap/src/main/resources/cli/listings.py]: can't the ``asyncio.get_event_loop().run_until_complete(get_response(query))`` be in this method?

File: cli-wrap/src/main/resources/cli/da-cli.py (hunk 5 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/da-cli.py]: I ran tests, and noticed that method for ./da-cli.py list white [PRODUCT_NAME:VERSION] is missing. I have no idea how i could overlook this so many times *shame*

File: cli-wrap/src/main/resources/cli/listings.py (hunk 3 above)
Comment (stanlyDoge) [inline cli-wrap/src/main/resources/cli/listings.py]: I am getting error that get_response(query) is not awaited anywhere

File: cli-wrap/src/main/resources/cli/da-cli.py (hunk 5 above)
Comment (janinko) [inline cli-wrap/src/main/resources/cli/da-cli.py]: But this doesn't have to do anything with WS, right? So at least put it into separate commit if not PR, please.

**Comment by janinko** [comment]
:+1:
